### DISCUSSION TITLE
Loop event inside Midi.

### DISCUSSION THREAD

**Discussion by RetroGamesClub** [vgmtrans/vgmtrans#269] [Q&A]
Is it possible to create Midi Loop Start/End event inside Midi files created from DS OST? Thank you.

**Comment by Wohlstand** [comment]
I'll append, there are some formats for loop points that were supported by various players:
- markers that contain "loopStart" and "loopEnd" keywords came from the music for Final Fantasy VII and were supported by many players and by Sekaiju (for example, my MIDI sequencer used at libADLMIDI and libOPNMIDI supports this loop format natively. I do use the same MIDI sequencer together with the FluidLite engine that is able to play MID+SF2/SF3 easily)
- Using CC111 controller came from the RPG Maker

There is also other loop tags format, but they are very uncommon and I don't suggest supporting them. FFVII and RPG Maker loop formats were most common.

**Reply by RetroGamesClub** [reply]
I usually use Falcosoft Midi player for midi playback of Nintendo DS .Mid + SF2 file; it would be great to have the midi generated from VGMTrams with the following loop start / loopend event. I Have to set this manually now but not always it's easy:
'loopStart' meta event on channel 1, Event: 176, Byte1: 116, Byte2: 0. 
'loopEnd' meta event Channel: 1, Event: 176, Byte1: 117, Byte2: 127.
Then always with above midi player i can decide the number of loop before the next song, but it would be great to have the loopstart/loopend event inside midi.
Aldo

**Reply by RetroGamesClub** [reply]
After about 6 month I tryed to download an update but still no Midi loop inside midi files generated from vtmtrans... Actually this is a prayer... Using Kode54 player on foobar I can listen to ds music which loop fine... I would like to have the same feature inside the midi files generated but still no loop information inside midi files generated :(

**Reply by peron93** [reply]
VGMtrans appears to focus more on reading information and less on conversion or manipulation.

**Comment by BLumia** [comment]
Would like +1 for this. It seems VGMTrans already recognized the `0x94` loop marker in SSEQ format but won't export it as a MIDI meta event. It could be handy if it can support such a feature.
